src/Model/Post/Engagement.php

```diff
@@ -331,17 +342,21 @@ private static function addResharers(string $text, int $uri_id): string
 		return $text;
 	}
 
-	public static function getMediaType(int $uri_id): int
+	public static function getMediaType(int $uri_id, int $quote_uri_id = null): int
 	{
 		$media = Post\Media::getByURIId($uri_id);
-		$type  = 0;
+		$type  = !empty($quote_uri_id) ? 16 : 0;
 		foreach ($media as $entry) {
 			if ($entry['type'] == Post\Media::IMAGE) {
 				$type = $type | 1;
 			} elseif ($entry['type'] == Post\Media::VIDEO) {
 				$type = $type | 2;
 			} elseif ($entry['type'] == Post\Media::AUDIO) {
 				$type = $type | 4;
+			} elseif ($entry['type'] == Post\Media::HTML) {
+				$type = $type | 8;
+			} elseif ($entry['type'] == Post\Media::ACTIVITY) {
+				$type = $type | 16;
```

Comment:
It's time to give all these bitmasks a name through constants.

Reply:
okay